src/app/zap-templates/templates/app/MatterIDL.zapt

```diff
@@ -0,0 +1,80 @@
+// This IDL was generated automatically by ZAP.
+// It is for view/code review purposes only.
+
+{{#chip_shared_structs}}
+{{#if struct_is_fabric_scoped}}
+[fabric_scoped_by=k{{asUpperCamelCase struct_fabric_idx_field}}]
+{{/if}}
+struct {{name}} {
+  {{#zcl_struct_items}}
+  {{type}} k{{asUpperCamelCase label}} = {{fieldIdentifier}};
+  {{/zcl_struct_items}}
+}
+
+{{/chip_shared_structs}}
+{{!TODO: consider #chip_clusters as iteration point as well.
+{{!      Unsure about the differences}}
+{{#all_user_clusters}}
+{{side}} cluster {{asUpperCamelCase name}} = {{code}} {
+  {{#zcl_enums}}
+  enum {{asUpperCamelCase name}} : {{type}} {
+    {{#zcl_enum_items}}
+    k{{asUpperCamelCase label}} = {{value}};
+    {{/zcl_enum_items}}
+  }
+
+  {{/zcl_enums}}
+  {{#chip_cluster_specific_structs}}
+  {{#if struct_is_fabric_scoped}}
+  [fabric_scoped_by=k{{asUpperCamelCase struct_fabric_idx_field}}]
+  {{/if}}
+  struct {{name}} {
+    {{#zcl_struct_items}}
+    {{type}} k{{asUpperCamelCase label}} = {{fieldIdentifier}};
+    {{/zcl_struct_items}}
+  }
+
+  {{/chip_cluster_specific_structs}}
+  {{#zcl_events}}
+  event {{asUpperCamelCase name}}({{priority}}) = {{code}} {
+    {{#zcl_event_fields}}
+    k{{asUpperCamelCase name}} = {{fieldIdentifier}},
+    {{/zcl_event_fields}}
+  }
+
+  {{/zcl_events}}
+  {{#chip_cluster_attributes}}
+  attribute({{#if isWritableAttribute}}readonly{{else}}writable{{/if}}{{#if isReportableAttribute}}, reportable{{/if}}) {{type}} {{asLowerCamelCase name}} = {{code}};
+  {{/chip_cluster_attributes}}
+  {{#chip_cluster_commands}}
+
+  request struct {{asUpperCamelCase name}}Request {
+    {{#chip_cluster_command_arguments}}
+    {{type}} {{label}};  {{!TODO: these should have an index here}}
```

Comment:
Should indicate which args are optional?

Reply:
Is there a term for it? I saw both nullable and optional in structs and other places, I still have to understand how those are defined.

Reply:
Updated for nullability and optional. Also added isList, but saw no proof that isList works (and I do not understand the difference between isList and isArray and never used  isArray).

Reply:
isList _might_ be defined for attributes being enumerated via chip_* helpers, maybe.

isArray should be defined for the various chip_* helpers.  I think.

Reply:
Will iterate over them - I expect we will converge in time to validate that what we see is what we expect from spec and behaviour.

I would keep this PR as a first step for now as it already does a lot.